src/fsharp/FSComp.txt

```diff
@@ -1146,7 +1146,7 @@ fscTooManyErrors,"Exiting - too many errors"
 3019,etPropertyHasSetterButNoCanWrite,"Property '%s' on provided type '%s' has CanWrite=false but GetSetMethod() returned a method"
 3020,etOneOrMoreErrorsSeenDuringExtensionTypeSetting,"One or more errors seen during provided type setup"
 3021,etUnexpectedExceptionFromProvidedTypeMember,"Unexpected exception from provided type '%s' member '%s': %s"
-3022,etUnsupportedConstantType,"Unsupported constant type '%s'"
+3022,etUnsupportedConstantType,"Unsupported constant type '%s'. Quotations provided by type providers can only contain simple constants. You may need to move a value declared outside a provided quotation literal to be a 'let' binding inside the quotation literal."
```

Comment:
Suggest

    Quotations provided by type providers can only contain simple constants. The implementation of the type provider may need to be adjusted by moving a value declared outside a provided quotation literal to be a 'let' binding inside the quotation literal.